web/src/settings_preferences.ts

```diff
@@ -152,23 +153,29 @@ function user_default_language_modal_post_render(): void {
                 setting_value,
             );
 
-            change_display_setting(
+            const $spinner = $("#settings_content").find(".general-settings-status").expectOne();
+            $spinner.fadeTo(0, 1);
+            loading.make_indicator($spinner, {text: settings_ui.strings.saving});
+            channel.patch({
+                url: "/json/settings",
                 data,
-                $("#settings_content").find(".general-settings-status"),
-                undefined,
-                undefined,
-                $t_html(
-                    {
-                        defaultMessage:
-                            "Saved. Please <z-link>reload</z-link> for the change to take effect.",
-                    },
-                    {
-                        "z-link": (content_html) =>
-                            `<a class='reload_link'>${content_html.join("")}</a>`,
-                    },
-                ),
-                true,
-            );
+                success(response_data) {
+                    const start_time = Date.now();
```

Comment:
This is the time when request is made not when response is received. See code for `settings_ui.do_settings_change`.